xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-rest/src/main/java/org/xwiki/annotation/rest/internal/AnnotationsRESTResource.java

```diff
@@ -162,39 +178,46 @@ protected AnnotationResponse postAnnotation(DocumentReference documentReference,
             // Initialize the context with the correct value.
             updateContext(documentReference);
 
-            String documentName = referenceSerializer.serialize(documentReference);
+            String documentName = this.referenceSerializer.serialize(documentReference);
 
             // check access to this function
-            if (!annotationRightService.canAddAnnotation(documentName, getXWikiUser())) {
+            if (!this.annotationRightService.canAddAnnotation(documentName, getXWikiUser())) {
                 throw new WebApplicationException(Status.UNAUTHORIZED);
             }
 
             // add the annotation
             Map<String, Object> annotationMetadata = getMap(request.getAnnotation());
-            annotationService.addAnnotation(documentName, request.getSelection(), request.getSelectionContext(),
+            this.annotationService.addAnnotation(documentName, request.getSelection(), request.getSelectionContext(),
                 request.getSelectionOffset(), getXWikiUser(), annotationMetadata);
+
             // and then return the annotated content, as specified by the annotation request
-            AnnotationResponse response = getSuccessResponseWithAnnotatedContent(documentName, request);
-            return response;
+            return getSuccessResponseWithAnnotatedContent(documentName, request);
         } catch (AnnotationServiceException | XWikiException e) {
             getLogger().error(e.getMessage(), e);
             return getErrorResponse(e);
         }
     }
 
-    /**
-     * Builds a simple map from a field collection.
-     *
-     * @param fields the collection of fields to build a map for
-     * @return a map of the fields, as string key and Object value pair
-     */
-    protected Map<String, Object> getMap(AnnotationFieldCollection fields)
+    private Map<String, Object> getMap(AnnotationFieldCollection annotation)
     {
-        Map<String, Object> metadataMap = new HashMap<String, Object>();
-        for (AnnotationField f : fields.getFields()) {
+        Map<String, Object> metadataMap = new HashMap<>();
+        for (AnnotationField f : annotation.getFields()) {
             metadataMap.put(f.getName(), f.getValue());
         }
 
+        // We perform conversion only if:
+        //   1. there is a RequiresHTMLConversion parameter
+        //   2. the value of this parameter is exactly "comment"
+        //   3. the syntax of comment is given in the parameters
```

Comment:
AFAIK the annotation module can be configured to use a different class (than ``XWikiComments``). So it's good to know, and maybe write down, that this code will work when annotations are saved as comments or, to be precise, when the annotation class has a ``comment`` property and that is used for storing the annotation text.